backend/src/apiserver/server/pipeline_server.go

```diff
@@ -610,6 +613,9 @@ func (s *PipelineServer) createPipelineVersion(ctx context.Context, pv *model.Pi
 
 	resp, err := s.httpClient.Get(pipelineUrl.String())
 	if err != nil || resp.StatusCode != http.StatusOK {
+		if err == nil {
```

Comment:
Looks like we can simplify the logic here into `if err == nil && resp.StatusCode != http.StatusOK`?